plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java

```diff
@@ -1042,14 +1044,37 @@ private void saveInputOutputSchema(String schemaFilePath) {
      * @param destinationFilePath path of the destination
      */
     private void copyInputFile(String originalFilePath, String destinationFilePath) {
-        File originalFile = new File(originalFilePath);
-        File copiedFile = new File(destinationFilePath);
+        String inputType = ((DataMapperDiagramEditor) workBenchPart.getSite().getWorkbenchWindow().getActivePage()
+                .getActiveEditor()).getInputSchemaType();
+
+        if (null != originalFilePath && null != destinationFilePath
+                && (inputType.equals("XML") || inputType.equals("JSON") || inputType.equals("CSV"))) {
+            File originalFile = new File(originalFilePath);
+            File copiedFile = new File(destinationFilePath);
+            try {
+                FileUtils.copyFile(originalFile, copiedFile);
+                assert (copiedFile).exists();
+                assert (Files.readAllLines(originalFile.toPath()).equals(Files.readAllLines(copiedFile.toPath())));
+            } catch (IOException e) {
+                log.error("Failed to create a copy of the original input file!");
```

Comment:
provide bit more context to the error message.

Reply:
Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063